relay-server/src/metrics_extraction/event.rs

```diff
@@ -49,15 +52,24 @@ impl Extractable for Span {
 /// If this is a transaction event with spans, metrics will also be extracted from the spans.
 pub fn extract_metrics(
     event: &mut Event,
-    spans_extracted: bool,
     config: CombinedMetricExtractionConfig<'_>,
+    sampling_decision: SamplingDecision,
     max_tag_value_size: usize,
-    span_extraction_sample_rate: Option<f32>,
-) -> Vec<Bucket> {
-    let mut metrics = generic::extract_metrics(event, config);
-    // If spans were already extracted for an event, we rely on span processing to extract metrics.
-    if !spans_extracted && sample(span_extraction_sample_rate.unwrap_or(1.0)) {
-        extract_span_metrics_for_event(event, config, max_tag_value_size, &mut metrics);
+    extract_spans: bool,
```

Comment:
Here, `extract_spans` replaces the combination of `spans_extracted` and `span_extraction_sample_rate`, and `sampling_decision` is only used for tagging purposes further in, correct?

Reply:
Correct, the caller determines whether spans should be extracted, which comprises three conditions: The sample rate, whether spans were extracted before, and whether span extraction is enabled at all. That last condition is new and required because we want to skip the root counter metric.

`sampling_decision` is there purely for tagging the root counter metric, right.